WordPress/gutenberg
# Remove Interactivity API experimental flag

## What?
I'm opening this pull request to be ready to remove the experimental flag for the Interactivity API and make the Navigation, File, and Image blocks use the Interactivity API by default. The idea is to ensure everything is ready for the moment we decide to remove the Experimental flag.

## Why?
At this moment, users can only test the Interactivity API and its functionalities if they activate the Experimental flag in the Gutenberg Experiments page.

## How?
These are the steps I took so far:
* I removed the conditional loading from the `load.php` file.
* I changed the `viewScript` field of the `block.json` of the Navigation, File, and Image blocks to read the `interactivity.js` file.
* I removed the old `view.js` files.
* I moved the directives directly to the `index.php` of the Image and File blocks. At this moment they are still using the Tag Processor, but I want to explore which is the best approach.
* I added some logic to ensure the JS files are only loaded when they are necessary. I want to explore the possibility of reusing the same function for all the blocks.
* I added some logic to load the directives conditionally (WIP)

## Testing Instructions
* Test that the Navigation, the File, and the Image lightbox blocks work as expected.
* Test that the scripts are loaded conditionally:
    * In the navigation block: when the overlay is set to off, and the submenus options to "open on click" and "show arrow" are disabled, scripts and  directives shouldn't be there.
    * In the file block: when the "Show inline embed" option is disabled, scripts and directives shouldn't be there.
    * In the image block: when the "Lightbox" is disabled from all the images, scripts and directives shouldn't be there.
